src/SqlConverters.cs

```diff
@@ -152,14 +150,19 @@ public async Task<IEnumerable<T>> ConvertAsync(SqlAttribute attribute, Cancellat
             /// The binding attribute that contains the name of the connection string app setting and query.
             /// </param>
             /// <returns></returns>
-            public virtual async Task<string> BuildItemFromAttributeAsync(SqlAttribute attribute)
+            public virtual async Task<string> BuildItemFromAttributeAsync(SqlAttribute attribute, ConvertType? type = null)
```

Comment:
Why is this optional? Looking at it the only places it's called all call TrackConvert so we should be sending the event with connection props for all of those, right?

Reply:
I made it optional for tests.

Reply:
That's generally not a pattern we should be using - instead you should be providing the appropriate mocks/passing the appropriate values as needed to simulate actual behavior as closely as possible.

Reply:
I know we have to simulate actual behavior but this is to collect telemetry, do we want data from tests as well?

Reply:
We will not be sending the actual event. Instead generally we would mock out the telemetry service so that it just no-ops any events being sent. (I don't know if this is actually being done currently, you may have to do it).

But the fact that this is used for telemetry is just an implementation detail, and not guaranteed to be the case in the future if we decide to use this value somewhere else so we shouldn't be making assumptions for that in the tests unless there's some reason we can't/shouldn't do that right now.